ts/Grid/Core/Querying/FilteringController.ts

```diff
@@ -0,0 +1,306 @@
+/* *
+ *
+ *  Grid Filtering Controller class
+ *
+ *  (c) 2020-2025 Highsoft AS
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ *  Authors:
+ *  - Dawid Dragula
+ *
+ * */
+
+'use strict';
+
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type {
+    StringCondition
+} from '../../../Data/Modifiers/FilterModifierOptions.js';
+
+import FilterModifier from '../../../Data/Modifiers/FilterModifier.js';
+import { FilteringOptions } from '../Options.js';
+
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * Class that manages one of the data grid querying types - filtering.
+ */
+class FilteringController {
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+    /**
+     * The simple conditions that are used in simple filtering.
+     */
+    private simpleConditions: StringCondition[];
+
+    private _modifier: FilterModifier;
+
+    /**
+     * The flag that indicates if the data should be updated because of the
+     * change in the filtering options.
+     */
+    public shouldBeUpdated: boolean = false;
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    constructor() {
+        this.simpleConditions = [];
+        this._modifier = new FilterModifier({
+            condition: {
+                operator: 'and',
+                conditions: this.simpleConditions
+            }
+        });
+    }
+
+
+    /* *
+    *
+    *  Functions
+    *
+    * */
+
+    public get modifier(): FilterModifier | undefined {
+        if (this.simpleConditions.length === 0) {
+            return;
+        }
+
+        return this._modifier;
+    }
+
+    /**
+     * Adds a new filtering condition to the specified column.
+     *
+     * @param columnId
+     * The column ID to filter in.
+     *
+     * @param options
+     * The filtering options.
+     */
+    public addFilterCondition(
+        columnId: string,
+        options: FilteringOptions
+    ): void {
+        const condition = this.mapOptionsToFilter(columnId, options);
```

Comment:
`mapOptionsToFilter` always returns a value, but the logic below suggests that it can be falsy. My point is that we should have the possibility to remove **only one** condition from `simpleConditions`, and that logic was implemented before. I'm not saying we should leave it as it was, just sharing a note that we should be aware of it.